src/main/java/team4/footwithme/resevation/service/ReservationServiceImpl.java

```diff
@@ -215,5 +216,24 @@ public void deleteParticipants(Long reservationId) {
         participantRepository.deleteAllInBatch(participants);
     }
 
+    /**
+     * 채팅방 상태 변경 API
+     * 채팅방장만 상태 변경 가능
+     */
+    @Transactional
+    @Override
+    public ReservationInfoResponse changeStatus(ReservationUpdateServiceRequest request, Member member) {
+        Reservation reservation = reservationRepository.findById(request.reservationId())
+                .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 예약입니다."));
+
+        if (!reservation.getMember().getMemberId().equals(member.getMemberId())) {
+            throw new IllegalArgumentException("예약한 사람만이 취소할 수 있습니다.");
+        }
+
+        reservation.updateStatus(request.status());
+
+        return ReservationInfoResponse.from(reservation);
+    }
```

Comment:
이게 정확하게 구분은 안가는데 "예외처리가 예약한 사람만이 취소 할 수 있습니다." 이렇게 되어있어서 헷갈리기는 합니다만, request에서는 status를 꺼내 쓸 수 있는데 예외는 취소만 할 수 있다고 보입니다.. 그래서 정확한 기능이 뭐인지는 모르겠습니다. 하여튼 결론적으로 말씀드리자면 취소만을 한다면 `updateStatus(request.status())` 보단 인자를 받지않는 `cancelReservation()` 과 같이 하면 좋을 것 같아요. 그게 아니라면 지금도 괜찮다고는 보입니다.

그리고 if문 안에는 리팩토링을 할 수 있지 않을까요? 디미터의 법칙도 생각해보시고, 가독성있게 만들 수 있다고 여겨집니다~

```suggestion
    public ReservationInfoResponse changeStatus(ReservationUpdateServiceRequest request, Member member) {
        Reservation reservation = reservationRepository.findById(request.reservationId())
                .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 예약입니다."));

        if (reservation.isReservationOwner(member.getMemberId())) {
            throw new IllegalArgumentException("예약한 사람만이 취소할 수 있습니다.");
        }

        reservation.updateStatus(request.status());

        return ReservationInfoResponse.from(reservation);
    }
```

대충 이런식으로요 이렇게 한다면 예외처리도 reservation으로 책임을 옮길 수도 있다고 여겨져요

사실은 궁극적으로 말하자면 updateStatus라는 메서드는 자신만이 할 수 있기 때문에 updateStatus안에 검증로직이 있는게 좋겠죠?
한번 생각해보세용

Reply:
정신을 반 놓고해서 그런지 위에 주석도 그렇고 쌩뚱맞은 문장을 작성했네요..

의도한건 예약한 사람만 예약 상태 수정을 할 수 있다는 건데 위에 메서드에서 그대로 복사해오고 수정을 안했습니다..

아래 내용은 참고해서 수정해보겠습니다! 감사합니다